### PR TITLE
Add month drill-down for judgments on TanzLII

### PR DESCRIPTION
**Pull request by nickmwangemi**
### Judgments page 
image

### Judgments filtered by year
Screenshot from 2023-08-24 08-44-06

### Judgments filtered by year and month
Screenshot from 2023-08-24 08-44-14

### Court registry judgments
Screenshot from 2023-08-24 08-44-20

### Court registry judgments filtered by year
Screenshot from 2023-08-24 08-44-31

### Court registry judgments filtered by year and month
Screenshot from 2023-08-24 08-44-37

### CODE REVIEW

**Comment by actlikewill** [comment]
@nickmwaengemi you might still be working on this, but on the court registry view, I'm not seeing all judgments on the all years view,

**Comment by nickmwangemi** [comment]
> @nickmwaengemi you might still be working on this, but on the court registry view, I'm not seeing all judgments on the all years view,

I've sorted this

File: peachjam/views/courts.py [hunk 1]
```diff
@@ -30,20 +35,42 @@ def get_queryset(self):
...
+                ] = f"{context['formatted_court_name']} - {context['month']} {self.kwargs['year']}"
+            else:
+                context[
+                    "formatted_court_name"
+                ] = f"{context['formatted_court_name']} - {self.kwargs['year']}"
+
+            context["all_months_url"] = reverse(
+                "court_year", args=[self.court.code, self.kwargs["year"]]
+            )
+            context["documents"] = self.grouped_judgments(self.get_base_queryset())
```
Comment (longhotsummer) [inline peachjam/views/courts.py:65]: this should always happen, not just for a year or month. Also, the grouping should happen on the main document list that is already in the context, thanks to the super class.

File: peachjam/views/courts.py [hunk 2]
```diff
@@ -123,35 +178,50 @@ def get_queryset(self):
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context["registry"] = self.registry
-        context["court"] = self.court
-        context["formatted_court_name"] = self.registry.name
+        context["grouped_judgments"] = self.grouped_judgments(self.get_base_queryset())
```
Comment (longhotsummer) [inline peachjam/views/courts.py:181]: this is unnecessary it's already being done by the parent class

File: peachjam/templates/peachjam/court_detail.html [hunk 3]
```diff
@@ -27,11 +27,16 @@ <h1 class="mt-4">{{ formatted_court_name }}</h1>
   {% include 'peachjam/_document_count.html' %}
   {% include 'peachjam/_registries.html' %}
   {% include 'peachjam/_years_list.html' with showallyears=True %}
+  {% if months %}
+    {% include 'peachjam/_months_list.html' with showallmonths=True %}
```
Comment (longhotsummer) [inline peachjam/templates/peachjam/court_detail.html:31]: I think the showallmonths (and the same above) is unnecessary.

File: peachjam/templates/peachjam/court_detail.html [hunk 4]
```diff
@@ -27,11 +27,16 @@ <h1 class="mt-4">{{ formatted_court_name }}</h1>
...
+  {% endif %}
 {% endblock %}
 {% block content %}
-  {% if documents %}
+  {% if documents and court_detail_view %}
```
Comment (longhotsummer) [inline peachjam/templates/peachjam/court_detail.html:35]: I'm pretty sure we always want to show a grouped judgments table, except when the list is sorted alphabetically and not by date, in which case grouping doesn't apply.

File: peachjam/templates/peachjam/_months_list.html [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+      <a class="{% if not month %} active{% endif %}"
+         href="{{ all_months_url }}">{% trans 'All months' %}</a>
+    </div>
+  {% endif %}
+  <div class="pl-0 mt-2 mb-3 align-items-center d-none d-lg-flex">
+    <i class="bi bi-calendar me-2"></i>
+    <ul class="year-nav mb-0 ms-2">
+      {% for m in months %}
+        <li>
+          <a class="{% if m.month == month %} active{% endif %}"
```
Comment (longhotsummer) [inline peachjam/templates/peachjam/_months_list.html:14]: if `m.month == month` then just render `m.month` without a tag around it.

File: peachjam/templates/peachjam/_months_list.html [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+             href="{{ m.url }}">{{ m.month }}</a>
+        </li>
+      {% endfor %}
+    </ul>
+  </div>
+{% endif %}
+{% if years %}
+  <select class="form-select mb-3 d-block d-lg-none"
```
Comment (longhotsummer) [inline peachjam/templates/peachjam/_months_list.html:22]: move into the if block above.

File: peachjam/templates/peachjam/_months_list.html (hunk 5 above)
Comment (nickmwangemi) [reply peachjam/templates/peachjam/_months_list.html:14]: By rendering without a tag, we'll lose the url.

File: peachjam/views/courts.py [hunk 7]
```diff
@@ -123,35 +177,49 @@ def get_queryset(self):
...
+        context["months"] = [
+            {
+                "url": reverse(
+                    "court_registry_month",
+                    args=[self.court.code, self.registry.code, year, item.month],
+                ),
+                "month": MONTHS[item.month],
+            }
+            for item in months
+        ]
```
Comment (longhotsummer) [inline peachjam/views/courts.py:217]: all months url missing

**Review by longhotsummer — APPROVED** [review]

**Comment by longhotsummer** [comment]
> ## Suspect Issues
> This pull request was deployed and Sentry observed the following issues:
> 
> * ‼️ **ProgrammingError: column peachjam_judgment.additional_citations does not exist** `/judgments/{code}/` [View Issue](https://lawsafrica.sentry.io/issues/4434090097/?referrer=github-pr-bot)
> * ‼️ **ProgrammingError: column peachjam_judgment.additional_citations does not exist** `/judgments/{code}/` [View Issue](https://lawsafrica.sentry.io/issues/4434142348/?referrer=github-pr-bot)
> 
> Did you find this useful? React with a 👍 or 👎

Strictly correct, but transient.

**Pull request merged**